internal/cmd/beta/network-interface/update/update_test.go

```diff
@@ -0,0 +1,286 @@
+	request := testClient.UpdateNic(testCtx, testProjectId, testNetworkId, testNicId)
+	request = request.UpdateNicPayload(fixturePayload())
+	for _, mod := range mods {
+		mod(&request)
+	}
+	return request
+}
+
+func fixturePayload(mods ...func(payload *iaas.UpdateNicPayload)) iaas.UpdateNicPayload {
+	var allowedAddresses []iaas.AllowedAddressesInner = []iaas.AllowedAddressesInner{
+		iaas.StringAsAllowedAddressesInner(utils.Ptr("1.1.1.1")),
+		iaas.StringAsAllowedAddressesInner(utils.Ptr("8.8.8.8")),
+		iaas.StringAsAllowedAddressesInner(utils.Ptr("9.9.9.9")),
+	}
+	payload := iaas.UpdateNicPayload{
+		AllowedAddresses: utils.Ptr(allowedAddresses),
+		Labels: utils.Ptr(map[string]interface{}{
+			"key": "value",
+		}),
+		Name:           utils.Ptr("testNic"),
+		NicSecurity:    utils.Ptr(true),
+		SecurityGroups: utils.Ptr([]string{testSecurityGroup}),
+	}
+	for _, mod := range mods {
+		mod(&payload)
+	}
+	return payload
+}
+func TestParseInput(t *testing.T) {
+	tests := []struct {
+		description   string
+		argValues     []string
+		flagValues    map[string]string
+		isValid       bool
+		expectedModel *inputModel
+	}{
+		{
+			description:   "base",
+			argValues:     fixtureArgValues(),
+			flagValues:    fixtureFlagValues(),
+			isValid:       true,
+			expectedModel: fixtureInputModel(),
+		},
+
+		{
+			description: "no values",
+			argValues:   fixtureArgValues(),
+			flagValues:  map[string]string{},
+			isValid:     false,
+		},
+		{
+			description: "network id missing",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, networkIdFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "network id invalid 1",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[networkIdFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "network id invalid 2",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[networkIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+		{
+			description: "allowed addresses missing",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, allowedAddressesFlag)
+			}),
+			expectedModel: fixtureInputModel(func(model *inputModel) {
+				model.AllowedAddresses = nil
+			}),
+			isValid: true,
+		},
+		{
+			description: "name to long",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[nameFlag] = "verylongstringwith66characterstotestthenameregexwithinthisunittest"
+			}),
+			isValid: false,
+		},
+		{
+			description: "name invalid",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[nameFlag] = "test?"
+			}),
+			isValid: false,
+		},
```

Comment:
Is an empty `name` valid? If not, it could also be included in the testing

Reply:
`name` is optional but an empty string "" is not valid. I will add this as well to the test.